treeherder/webapp/api/performance_data.py

```diff
@@ -751,6 +753,31 @@ def _filter_out_retriggers(serialized_data: List[dict]) -> List[dict]:
         return serialized_data
 
 
+class PerformanceAlertSummaryTasks(generics.ListAPIView):
+    serializer_class = PerformanceAlertSummaryTasksSerializer
+    queryset = None
+
+    def list(self, request):
+        query_params = PerfAlertSummaryTasksQueryParamSerializer(data=request.query_params)
+        if not query_params.is_valid():
+            return Response(data=query_params.errors, status=HTTP_400_BAD_REQUEST)
+
+        alert_summary_id = query_params.validated_data['alert_summary_id']
```

Comment:
Based on the tests I believe this line should be changed because the param is `?id=` not `?alert_summary_id=`
```suggestion
        alert_summary_id = query_params.validated_data['id']
```
Have I misunderstood something?

Reply:
The parameter requires an `?id=`, not `?alert_summary_id=` :)
The `alert_summary_id` is just an alias in the code base.

That's the serialize code[0]:
```
class PerfAlertSummaryTasksQueryParamSerializer(serializers.Serializer):
    id = serializers.IntegerField(source='alert_summary_id')
```
The `id` is for the parameter and `alert_summary_id` is used as an alias in the code base.

This might be a little confusing, so I'll change `alert_summary_id` to `id` :)

[0]
https://github.com/mozilla/treeherder/pull/7717/files#diff-1476e733bc1922c0d51a5d9cbd8d061b13b67ee776b0565f4d6d4d43da10f8b3R401